src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/ExtendedFortisEvent.scala

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.partnercatalyst.fortis.spark.analyzer
+
+import java.util.UUID
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{Analysis, Details, FortisEvent, Location}
+
+case class ExtendedFortisEvent[T](
+  details: ExtendedDetails[T],
+  analysis: Analysis
+) extends FortisEvent
+
+case class ExtendedDetails[T](
+  id: UUID,
+  createdAtEpoch: Long,
+  body: String,
+  title: String,
+  publisher: String,
+  sourceUrl: String,
+  sharedLocations: List[Location] = List(),
+  original: T
```

Comment:
Again, the requirement already exists, even if we haven't hit the eventuality that show-stopping conflicts between pipeline requirements can occur with the current approach. I wouldn't waste my own time if I didn't see a clear issue.

I completely agree that we've wasted time, but the time wasn't wasted through this refactor, but rather at the point when we'd jumped into an initial design without research and discussion. That's on all of us as a team. FWIW, this PR makes extensive use of the implementation patterns (if not verbatim copies) introduced in #22.

The purpose of this PR's comment section is to discuss its implementation and to communicate design decisions / transfer understanding. If it has not yet served that purpose, I encourage you to add more discussion points. I'm happy to explain any choices I've made here, and to ideally convince you that this is a good way forward as well.